### PR TITLE
Add Mode Eighty M80v2 PCB support

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Adds support for Mode Design's Mode Eight M80v2 PCBs.

The PR also moves the v1 PCBs firmware files to a  `v1` folder, whereas the v2 firmware is in a `v2` folder. The readme's were updated with compiling and flashing instructions.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/mode/m80v2/m80v2s/rules.mk [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# MCU name
+MCU = STM32F072
```
Comment (fauxpark) [inline keyboards/mode/m80v2/m80v2s/rules.mk:2]:
```suggestion
MCU = STM32F072

# Bootloader selection
BOOTLOADER = stm32-dfu
```

File: keyboards/mode/m80v2/m80v2s/rules.mk [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes      # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes          # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no          # Audio output
+# Enter lower-power sleep mode when on the ChibiOS idle thread
```
Comment (fauxpark) [inline keyboards/mode/m80v2/m80v2s/rules.mk:16]:
```suggestion

# Enter lower-power sleep mode when on the ChibiOS idle thread
```

File: keyboards/mode/m80v2/m80v2h/rules.mk [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# MCU name
+MCU = STM32F072
```
Comment (fauxpark) [inline keyboards/mode/m80v2/m80v2h/rules.mk:2]:
```suggestion
MCU = STM32F072

# Bootloader selection
BOOTLOADER = stm32-dfu
```

File: keyboards/mode/m80v2/m80v2h/rules.mk [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes      # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = yes          # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no          # Audio output
+# Enter lower-power sleep mode when on the ChibiOS idle thread
```
Comment (fauxpark) [inline keyboards/mode/m80v2/m80v2h/rules.mk:16]:
```suggestion

# Enter lower-power sleep mode when on the ChibiOS idle thread
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by BradMello** [comment]
@fauxpark Please review when you get a chance. Thanks

**Review by fauxpark — APPROVED** [review]

**Pull request merged**